extensions/interactions/GraphInput/directives/GraphInput.js

```diff
@@ -191,18 +191,20 @@ oppia.directive('oppiaResponseGraphInput', [
       controller: ['$scope', '$attrs', function($scope, $attrs) {
         $scope.graph = HtmlEscaperService.escapedJsonToObj($attrs.answer);
 
-        $scope.VERTEX_RADIUS = graphDetailService.VERTEX_RADIUS;
-        $scope.EDGE_WIDTH = graphDetailService.EDGE_WIDTH;
-        $scope.GRAPH_INPUT_LEFT_MARGIN = GRAPH_INPUT_LEFT_MARGIN;
-
-        $scope.getDirectedEdgeArrowPoints = function(index) {
-          return graphDetailService.getDirectedEdgeArrowPoints(
-            $scope.graph, index);
-        };
-
-        $scope.getEdgeCentre = function(index) {
-          return graphDetailService.getEdgeCentre($scope.graph, index);
-        };
+        if ($scope.graph.edges) {
```

Comment:
Can you explain the reason behind this? (Perhaps in a codebase comment if it's important for future devs to know.)

Reply:
For this and the comments below, it seems that when the learner input led to a new card, the card was displayed before it was recorded in PlayerTranscriptService due to the various timeouts and new sevice calls (I think). So, the interaction used the previous interaction's learner input for a shortwhile leading to console errors. Though this doesn't affect user experience because as soon as transcript was updated, interaction does load correctly with no user facing errors.

Reply:
Thanks for the explanation. That seems like a really indirect fix. Can you do two things please, in each of the files:

1) Add a comment above the check explaining why this check is there and that it should really be removed once the root cause of this issue is fixed.

2) File issues to actually fix the root cause.

Also, why didn't this happen before (or was it already happening before and you just found out about it now)?

Reply:
Ok, will add comment and file issue.
This happened before only for numeric input,for which I had already added a comment, but now it is happening for graph and image as well.
I played through the all_interactions exp and only for these 3 interactions did I find this. Everything else worked fine.

Reply:
Filed [issue](https://github.com/oppia/oppia/issues/5518).